### ISSUE TITLE
Make expect(false) work as a failing assertion

### ISSUE THREAD

**Issue by fulldecent** [chaijs/chai#1408]
If there is going to be magic like `expect.to.be.true` (without the `()` on true), then it is surprising that `expect(false)` is not equivalent to `expect(false).to.equal(true)`.

Your test will pass if you have `expect(false)` inside. And this is dangerous.

**Comment by sukrosono** [comment]
Yes, priorly `expect(evaluation)` is more dangerous, let's hear another input.

**Comment by koddsson** [comment]
Hey @fulldecent and @sukrosono; I'm having trouble figuring out what the issue is. Can you write up the issue in an `Expected results` / `Actual results` format so that it's a bit more clear what you are expecting `chai` to do and what it's doing? Any additional information about why the behaviour is expected of you would be helpful as well.

**Comment by fulldecent** [comment]
# Test case:

```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>Chai expect false example</title>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/mocha/9.1.3/mocha.min.css">
</head>
<body>
  <div id="mocha"></div>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mocha/9.1.3/mocha.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/chai/4.3.4/chai.min.js"></script>  
  <script>
    // set up mocha with bdd interface
    mocha.setup('bdd');
    const expect = chai.expect;
    
    // define a test suite
    describe('chai expect false', function() {
      it('Zero should equal one', function() {
        expect(0 === 1)
      });
    });
    
    // run mocha tests
    mocha.run();
  </script>
</body>
</html>
```

# Expected outcome:

> Zero should equal one
>
> Test case fails ❌ 

# Actual outcome

> Zero should equal one
>
> Test case passes, asserting that zero equals one

**Comment by jdmarshall** [comment]
does this pass the chai linter?

I'm not sure I like this one. As a long time chai user I believe I would ask my coworker to fix their PR if they tried to add this pattern to a test suite.
